src/dftbp/common/memman.F90

```diff
@@ -30,4 +80,59 @@ pure function incrmntOfArray(currentSize)
 
   end function incrmntOfArray
 
+
+  !> Allocate an array aligned to the n byte boundary (typically used for aligning to 64 bytes)
+  subroutine TAlignedArray_allocateAligned(this, size, alignment)
+
+    !> Instance of the aligned array type
+    class(TAlignedArray), intent(inout) :: this
+
+    !> Lower and upper bound for alignment
+    integer, intent(in) :: size
+
+    !> Alignment (default: 64 bytes)
+    integer, intent(in), optional :: alignment
+
+    if (this%isAllocated) then
+      call error("Aligned array is already allocated")
+    end if
+
+    @:ASSERT(size > 0)
+
+    if (present(alignment)) then
+      @:ASSERT(alignment >= dp)
+      @:ASSERT(mod(alignment, dp) == 0)
+
+      this%alignment = alignment
+    end if
+
+    if (posixMemalign(this%memoryPointer, int(this%alignment, kind=c_size_t),&
+          & int(size * dp, kind=c_size_t)) /= 0) then
+      call error("Error during allocation of aligned array")
+    end if
+
+    this%size = size
+    call c_f_pointer(this%memoryPointer, this%array, [size])
+
+    this%isAllocated = .true.
+
+  end subroutine TAlignedArray_allocateAligned
+
+
+  !> Clean-up an aligned array by deallocating it
+  subroutine TAlignedArray_finalize(this)
```

Comment:
Note: if someone creates an array of `type(TAlignedArray)` , the destructor will not be called when it goes out of scope! You should make it `elemental impure` to have it invoked for any ranks.